lib/common/patchset_display.c

```diff
@@ -132,27 +147,104 @@ xml_show_patchset_v1(pcmk__output_t *out, const xmlNode *patchset)
+
         if (is_first) {
             is_first = false;
         } else {
-            out->info(out, " --- ");
+            rc = pcmk__output_select_rc(rc, out->info(out, " --- "));
         }
     }
 
     is_first = true;
     added = find_xml_node(patchset, "diff-added", FALSE);
     for (child = pcmk__xml_first_child(added); child != NULL;
          child = pcmk__xml_next(child)) {
-        xml_show_patchset_v1_recursive(out, "+ ", child, 0,
-                                       options|pcmk__xml_fmt_diff_plus);
+        int temp_rc = xml_show_patchset_v1_recursive(out, "+ ", child, 0,
+                                                     options
+                                                     |pcmk__xml_fmt_diff_plus);
+        rc = pcmk__output_select_rc(rc, temp_rc);
+
         if (is_first) {
             is_first = false;
         } else {
-            out->info(out, " +++ ");
+            rc = pcmk__output_select_rc(rc, out->info(out, " +++ "));
         }
     }
+
+    return rc;
+}
+
+/*!
+ * \internal
+ * \brief Output an XML patchset (format 1)
+ *
+ * This function parses an XML patchset (an \p XML_ATTR_DIFF element and its
+ * children) into a user-friendly combined diff output.
+ *
+ * \param[in,out] out       Output object
+ * \param[in]     patchset  XML patchset to output
+ *
+ * \return Standard Pacemaker return code
+ */
+PCMK__OUTPUT_ARGS("xml-patchset-v1", "xmlNode *")
+static int
+xml_patchset_v1_default(pcmk__output_t *out, va_list args)
+{
+    xmlNode *patchset = va_arg(args, xmlNode *);
+
+    return xml_show_patchset_v1_common(out, patchset, pcmk__xml_fmt_pretty);
+}
+
+/*!
+ * \internal
+ * \brief Output an XML patchset (format 1)
+ *
+ * This function parses an XML patchset (an \p XML_ATTR_DIFF element and its
+ * children) into a user-friendly combined diff output.
+ *
+ * \param[in,out] out       Output object
+ * \param[in]     patchset  XML patchset to output
+ *
+ * \return Standard Pacemaker return code
+ */
+PCMK__OUTPUT_ARGS("xml-patchset-v1", "xmlNode *")
+static int
+xml_patchset_v1_log(pcmk__output_t *out, va_list args)
+{
+    xmlNode *patchset = va_arg(args, xmlNode *);
+
+    uint32_t options = pcmk__xml_fmt_pretty;
+
+    if (pcmk__output_get_log_level(out) < LOG_DEBUG) {
+        options |= pcmk__xml_fmt_diff_short;
+    }
+    return xml_show_patchset_v1_common(out, patchset, options);
+}
+
+/*!
+ * \internal
+ * \brief Output an XML patchset (format 1)
+ *
+ * This function outputs an XML patchset (an \p XML_ATTR_DIFF element and its
+ * children) without modification.
+ *
+ * \param[in,out] out       Output object
+ * \param[in,out] patchset  XML patchset to output
+ *
+ * \return Standard Pacemaker return code
+ */
+PCMK__OUTPUT_ARGS("xml-patchset-v1", "xmlNode *")
+static int
+xml_patchset_v1_xml(pcmk__output_t *out, va_list args)
+{
+    xmlNode *patchset = va_arg(args, xmlNode *);
+
+    pcmk__output_xml_add_node_copy(out, patchset);
```

Comment:
The benefit of a CDATA block is the API schema can just use `<text/>` for it rather than need to know the separate schema for what's being added.

Reply:
We already have something like an `any.rng`, and I was considering using that as an alternative to CDATA... somewhere! I wrote it, I thought as part of the rule schema update, but I can't find it in the source tree.

Reply:
The mystery will remain unsolved. No real reason to make one now (rather than using CDATA) if we don't already have one.

Reply:
I was using the `any-element` schema in this `diff-2.29.rng` schema, and I definitely linked to the gist within some other PR a couple months ago (don't know which one). But those schemas themselves haven't been part of any PR yet.